### PR TITLE
Unit tests for `shell_env_vars` logic

### LINKED ISSUE #386: FIxup bugs from #383 - Add end to end environment variable test script
So we don't loose track of it - need a followup PR to fixup bugs introduced in #383:

These also don't look like they match the schema names.
- [x] The end to end test script I mentioned would probably help catch this.

_Originally posted by @bpkroth in https://github.com/microsoft/MLOS/pull/383#discussion_r1222268700_

### PR DESCRIPTION
**Pull request by motus**
Summary of changes:
* Add unit tests for `LocalEnv` to make sure that we pass environment variables into a local script correctly.
* Split unit tests for `LocalEnv` into three separate files for maintainability.

P.S. Turned out, we pass the environment variables into a local script correctly - we just have to use the cmd syntax `%VAR%` on Windows instead of `$VAR`. We now have unit tests for both cases.

-----------
> It looks like `local_exec` service fails to set the environment variables on Windows.
> This PR adds a unit test to reproduce the problem.
> We need to debug the issue and push the fix into this or other PR.
> 
> P.S. `local_env_test.py` has too many tests now, so in this PR we split it into three separate files that have tests for the results, telemetry, and environment variables, respectively.

~~**NOTE:** Merge after #483~~

Closes #386

### CODE REVIEW

**Comment by bpkroth** [comment]
> It looks like `local_exec` service fails to set the environment variables on Windows. We need to debug the issue.
> 
> **NOTE:** Merge after #483
> 
> Closes #386

I was actually seeing cases where we don't propagate environment variables to Linux shell scripts either.
I'll cook a smaller example up and we can address that in a different PR though.
Thanks for adding the tests!

File: mlos_bench/mlos_bench/tests/environments/local_env_test.py [hunk 1]
```diff
@@ -66,3 +68,132 @@ def test_local_env(tunable_groups: TunableGroups) -> None:
...
+                "other_arg": 222,  # NOT passed into "shell_env_params"
+            },
+            "tunable_params": ["kernel"],
+            "shell_env_params": [
+                "const_arg",                # From "const_arg"
+                "kernel_sched_latency_ns",  # From "tunable_params"
+            ],
+            "run": [
+                "echo const_arg,other_arg,unknown_arg,kernel_sched_latency_ns > output.csv",
+                "echo $const_arg,$other_arg,$unknown_arg,$kernel_sched_latency_ns >> output.csv",
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/environments/local_env_test.py:183]: Would be good to also test this in a composite nested env.
That's where I've run into issues previously.

Comment (motus) [reply mlos_bench/mlos_bench/tests/environments/local_env_test.py]: yes, I plan to add many more tests for the composite environment

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/environments/local_env_test.py]: I assume that's for a future PR then?

**Review by bpkroth — APPROVED** [review]

File: mlos_bench/mlos_bench/tests/environments/local_env_test.py (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/environments/local_env_test.py]: yes

**Pull request merged**